src/ValidateOrder.php

```diff
@@ -118,32 +118,32 @@ public function validateOrder($payload)
                 }
             }
 
-            $psCheckoutCartCollection = new \PrestaShopCollection('PsCheckoutCart');
-            $psCheckoutCartCollection->where('id_cart', '=', (int) $payload['cartId']);
+            /** @var \Ps_checkout $module */
+            $module = \Module::getInstanceByName('ps_checkout');
+
+            /** @var \PrestaShop\Module\PrestashopCheckout\Repository\PsCheckoutCartRepository $psCheckoutCartRepository */
+            $psCheckoutCartRepository = $module->getService('ps_checkout.repository.pscheckoutcart');
 
             /** @var \PsCheckoutCart|false $psCheckoutCart */
-            $psCheckoutCart = $psCheckoutCartCollection->getFirst();
+            $psCheckoutCart = $psCheckoutCartRepository->findOneByCartId((int) $payload['cartId']);
 
             if (false === $psCheckoutCart) {
                 $psCheckoutCart = new \PsCheckoutCart();
                 $psCheckoutCart->id_cart = (int) $payload['cartId'];
                 $psCheckoutCart->paypal_intent = $paypalOrder->getOrderIntent();
                 $psCheckoutCart->paypal_order = $response['body']['id'];
                 $psCheckoutCart->paypal_status = $response['body']['status'];
-                $psCheckoutCart->add();
+                $psCheckoutCartRepository->save($psCheckoutCart);
             } else {
                 $psCheckoutCart->paypal_order = $response['body']['id'];
                 $psCheckoutCart->paypal_status = $response['body']['status'];
-                $psCheckoutCart->update();
+                $psCheckoutCartRepository->save($psCheckoutCart);
             }
 
```

Comment:
duplicate line here : in all case we do this :
$psCheckoutCart->paypal_order = $response['body']['id'];
$psCheckoutCart->paypal_status = $response['body']['status'];
$psCheckoutCartRepository->save($psCheckoutCart);